### PR TITLE
[Couchbase VectorStore] Add RecordManager

### PR DESCRIPTION
**Pull request by JJK801**
Hi Flowise,

This PR adds RecordManager capability to the existing Couchbase vector store node.

Related https://github.com/FlowiseAI/Flowise/pull/3093

### CODE REVIEW

**Comment by HenryHengZJ** [comment]
would you be able to pull the latest changes since this [PR](https://github.com/FlowiseAI/Flowise/pull/3093) was merged

**Comment by JJK801** [comment]
@HenryHengZJ Done

**Comment by JJK801** [comment]
@prajwal-pai77 @HenryHengZJ I'm wondering how to handle the `vectorStoreName` key for indexing, any advice ?

File: packages/components/nodes/vectorstores/Couchbase/Couchbase.ts [hunk 1]
```diff
@@ -169,8 +178,28 @@ class Couchbase_VectorStores implements INode {
...
+                    await recordManager.createSchema()
+
+                    const res = await index({
+                        docsSource: finalDocs,
+                        recordManager,
+                        vectorStore,
+                        options: {
+                            cleanup: recordManager?.cleanup,
+                            sourceIdKey: recordManager?.sourceIdKey ?? 'source',
+                            vectorStoreName: [bucketName, scopeName, collectionName].join('_')
```
Comment (JJK801) [inline packages/components/nodes/vectorstores/Couchbase/Couchbase.ts:193]: @prajwal-pai77 @HenryHengZJ I'm wondering how to handle the vectorStoreName key for indexing, any advice ?

should we include `indexName` ?

**Comment by HenryHengZJ** [comment]
Is couchbase supporting record manager from langchainJS? Have you tested it?

**Comment by JJK801** [comment]
> Is couchbase supporting record manager from langchainJS? Have you tested it?

I didn't test it already, but AFAIK the record manager is agnostic of the vector store, its just a DB indexing chunks states.

**Comment by HenryHengZJ** [comment]
> > Is couchbase supporting record manager from langchainJS? Have you tested it?
> 
> I didn't test it already, but AFAIK the record manager is agnostic of the vector store, its just a DB indexing chunks states.

Ah okay. I'll test it. There is a reason why record manager is not added to all vector stores. The addvectors and delete methods need to be able to take in ids as parameters in order to be able to use record manager. You can see the example of Qdrant that extends those functions for that

**Comment by HenryHengZJ** [comment]
@JJK801 for some reasons i just can't seem to have Couchbase Vector Store working in order to test this, have you got any success with that?

**Comment by JJK801** [comment]
Hi @HenryHengZJ sorry for the delay, i didn't had time to test it and i will finally postpone the usage of couchbase in my project.

Feel free to close it if needed.

**Comment by HenryHengZJ** [comment]
Closing for now

**Pull request closed without merging**